### PR TITLE
Add Mimir alertmanager in control center

### PR DESCRIPTION
**Pull request by muzammil360**

### CODE REVIEW

File: gitops/applications/base/monitoring-post-config/mimir-alertmanager.yaml [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    serviceAccount:
+      name: default
+  passwordPolicy: |
+    length = 5
+      rule "charset" {
+        charset = "-"
+        min-chars = 1
+      }
+---
+# NOTE: the secret generated by RandomSecret resources in this file are not used as is.
```
Comment (dfry) [inline gitops/applications/base/monitoring-post-config/mimir-alertmanager.yaml:19]: have you tested that the manually entered values do bot get overwritten by the random secret on a subsequent argocd sync operation?
overall I am confused why you define both random secrets and external secrets.

File: gitops/applications/base/monitoring-post-config/mimir-alertmanager.yaml [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+            receiver: ${ARGOCD_ENV_mimir_alerts_notification_receiver}
+            group_by: ['cluster', 'job', 'alertname']
+            group_wait: 30s
+            group_interval: 5m
+            repeat_interval: 24h
+          receivers:
+            - name: 'blackhole'
+            - name: 'slack'
+              slack_configs:
+                - api_url: {{.webhook}}
```
Comment (sijo5722-2021) [inline gitops/applications/base/monitoring-post-config/mimir-alertmanager.yaml:96]: Does api_url sourced by webhook needs to be configured as a random secret ?

File: gitops/applications/base/monitoring-post-config/mimirtool-cronjob.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  name: mimir-rules-alerts-sync-job
+spec:
+  schedule: "*/1 * * * *"  # Runs every minute
```
Comment (dfry) [inline gitops/applications/base/monitoring-post-config/mimirtool-cronjob.yaml:6]: these rules will change all the time?

File: gitops/applications/base/monitoring/values-mimir.yaml [hunk 4]
```diff
@@ -16,12 +16,30 @@ mimir:
...
+      alertmanager_url: http://central-monitoring-grafana-mimir-gateway/alertmanager
     limits:
       max_global_series_per_user: ${ARGOCD_ENV_mimir_max_global_series_per_user}
       ingestion_rate: ${ARGOCD_ENV_mimir_ingestion_rate}
       ingestion_burst_size: ${ARGOCD_ENV_mimir_ingestion_burst_size}
       compactor_blocks_retention_period: ${ARGOCD_ENV_mimir_retention_period}
       max_label_names_per_series: ${ARGOCD_ENV_mimir_max_label_names_per_series}
+    common:
+      storage:
+        backend: s3
```
Comment (dfry) [inline gitops/applications/base/monitoring/values-mimir.yaml:29]: why do we have s3 aws config here?

File: gitops/applications/base/monitoring-post-config/mimir-alertmanager.yaml (hunk 2 above)
Comment (muzammil360) [reply gitops/applications/base/monitoring-post-config/mimir-alertmanager.yaml:96]: No. I just want to create a field with the name `webhook`. Ideally the initial value should be a static value e.g `update_me`. But I couldn't find a way to make this key with static value. Any suggestions around that?

File: gitops/applications/base/monitoring/values-mimir.yaml (hunk 4 above)
Comment (muzammil360) [reply gitops/applications/base/monitoring/values-mimir.yaml:29]: @dfry what do you mean? Why shouldn't they be here?

File: gitops/applications/base/monitoring-post-config/mimir-alertmanager.yaml (hunk 1 above)
Comment (muzammil360) [reply gitops/applications/base/monitoring-post-config/mimir-alertmanager.yaml:19]: > have you tested that the manually entered values do bot get overwritten by the random secret on a subsequent argocd sync operation?
@dfry yes. I have tested. I don't provide `refreshPeriod`, therefore random secret doesn't update the value after the creation of secret. However if we deleted the CR, then the value will be over-written in the vault. But we don't need to solve this problem. 

> overall I am confused why you define both random secrets and external secrets.
`RandomSecret` creates a secret in vault. That's it. It doesn't need to be random. But i don't have any easy way to make a static one either. 

`ExternalSecret` pulls the secret data from vault and formats alertmanager's configuration file.

Comment (muzammil360) [reply gitops/applications/base/monitoring-post-config/mimir-alertmanager.yaml:19]: > have you tested that the manually entered values do bot get overwritten by the random secret on a subsequent argocd sync operation?

@dfry yes. I have tested. `refreshPeriod` is the field which forces value refresh. If we don't provide it, the secret is only created once.

File: gitops/applications/base/monitoring/values-mimir.yaml (hunk 4 above)
Comment (dfry) [reply gitops/applications/base/monitoring/values-mimir.yaml:20]: what would happen if you ran it on prem and those env vars didn't exist?

Comment (muzammil360) [reply gitops/applications/base/monitoring/values-mimir.yaml:20]: > what would happen if you ran it on prem and those env vars didn't exist?

@dfry . this is not really AWS S3. This is using ceph object store.

**Pull request merged**